recursion/gnark/go.mod

```diff
@@ -1,19 +1,41 @@
 module github.com/succinctlabs/sp1-recursion-gnark
 
-go 1.20
+go 1.22.1
```

Comment:
note: I'm bumping this just because Go added some nice http features in std lib, but if it's a problem because of Gnark audits lmk

Reply:
no problem